dejavu/database.py

```diff
@@ -55,9 +58,18 @@ def set_song_fingerprinted(self, sid):
         song.fingerprinted = True
         self.session.commit()
 
-    def get_songs(self):
-        """Returns all fully fingerprinted songs in the database."""
-        return self.session.query(Song).filter(Song.fingerprinted)
+    # def get_songs(self): - don't cache all of this!
```

Comment:
What do you mean by this comment?

Reply:
in trying to parallelize the run process across many machines, I found that it was running a database scan (select * songs, return id) locally for each instance.  Instead of doing that by *default* on *every load*, I modified the song check to only the relevant parts of code that were trying to extract and ingest a new one (see new use of `get_song_by_hash`).